torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -779,14 +788,14 @@ def get_state_dict_type(module: nn.Module) -> StateDictSettings:
                     f"Got {submodule_settings} and {state_dict_settings}."
                 )
 
-            # If device_mesh is passed in when initalizing FSDP, we automatically turn the
-            # _use_dtensor flag to be true for ShardedOptimStateDictConfig().
-            if (
-                getattr(module, "device_mesh", None)
-                and state_dict_settings.state_dict_type
-                == StateDictType.SHARDED_STATE_DICT
-            ):
-                state_dict_settings.optim_state_dict_config._use_dtensor = True
+                # If device_mesh is passed in when initalizing FSDP, we automatically turn the
+                # _use_dtensor flag to be true for ShardedOptimStateDictConfig().
+                if (
+                    getattr(submodule, "device_mesh", None)
+                    and submodule_settings.state_dict_type
+                    == StateDictType.SHARDED_STATE_DICT
+                ):
+                    submodule_settings.optim_state_dict_config._use_dtensor = True
```

Comment:
This seems repeating exactly L772. Can this be a sanity check function to ensure consistency?